### PR TITLE
Improve project organization

### PR DESCRIPTION
**Pull request by shykes**
Proposed improvements to project organization.

This organization change has 3 goals:

1) Make the project more open and accessible
2) Make the project more scalable
3) Do this in an incremental way, without unnecessary refactoring

The actual organization change is covered in 6a99b8dea935be9228a47c84bcbfb7491e6e5584. Cosmetic changes are discussed in #9177 and #9178

### CODE REVIEW

**Comment by vbatts** [comment]
@shykes next steps? and DCO marker? ;-)

File: MAINTAINERS [hunk 1]
```diff
@@ -1,9 +1,456 @@
...
+		people = [
+			"backjlack",
+			"crosbymichael",
+			"erikh",
+			"icecrime",
+			"jfrazelle",
+			"lkd4d"
+			"tibor",
+			"vbatts"
+			"vieux",
```
Comment (tiborvass) [inline MAINTAINERS]: vbatts misses comma, and vieux should not have one

File: MAINTAINERS [hunk 2]
```diff
@@ -1,9 +1,462 @@
...
+
+
+
+		people = [
+			"backjlack",
+			"crosbymichael",
+			"erikh",
+			"icecrime",
+			"jfrazelle",
+			"lkd4d"
```
Comment (icecrime) [inline MAINTAINERS]: It's "lk4d4"!

File: MAINTAINERS [hunk 3]
```diff
@@ -1,9 +1,462 @@
...
+			people = [
+				"jess",
+				"crosbymichael"
+			]
+
+	[Org.Subsystems]
+
+	# As the project grows, it gets separated into well-defined subsystems. Each subsystem
+	# has a dedicated group of maintainers, which are responsible for its quali
+	# dedicated to that subsytem. This "cellular division" is the primary mechanism
```
Comment (icecrime) [inline MAINTAINERS]: Something wrong here (truncated word "quali" unrelated to the end of the sentence).

**Comment by tiborvass** [comment]
Overall LGTM :)

**Comment by icecrime** [comment]
Same here, I don't see any remaining issues: LGTM :-1:

**Comment by unclejack** [comment]
LGTM

**Comment by jfrazelle** [comment]
LGTM

**Comment by shykes** [comment]
Ok, after much discussion on irc, I think we're ready to merge.

* Added missing people
* Added missing subsystems
* Officialized de-facto core maintainers
* Removed the vague notion of "lead"
* Clarified the roles of Chief Architect, Chief Maintainer, Chief Operator

Even merged, this doesn't end the conversation. The goal is to formalize a structure which is already mostly in place informally. This is a necessary starting point for attacking more ambitious upgrades to the organization.

The big topic which remains to be addressed: how do we transform from a tool-centric organization to a platform-centric organization? Docker is no longer a single tool, but a collection of tools unified under a platform. We want this platform to be composable (each tool should be useful standalone, and replaceable by another), but also well integrated: the user experience should feel like more than the sum of the parts. To achieve this, we need to address *integration* at many levels. More on this topic soon!

**Comment by shykes** [comment]
@cyphar it's mostly formalizing what is already in place, but also incorporating de-facto rules that have appeared organically but weren't documented. For example, the concept of "core maintainers" which are the de facto owners of everything which isn't a clearly separated subsystem - that was already the case but not clearly documented. This file also deprecates the more fine-grained MAINTAINERS files - because in practice we were ignoring them often.

The other important change is that I am reducing the role of the bdfl, and creating 3 new leadership roles: Chief Architect, Chief Maintainer, Chief Operator. That should help reduce the dependency on bdfl for running the project, while keeping it as a guarantee for the long-term direction and governance of the project.

**Pull request merged**